### PR TITLE
initial implementation

### PR DESCRIPTION
**Pull request by troshkin**

### CODE REVIEW

File: src/Lykke.Job.BitcoinGold/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,4 @@
+FROM microsoft/aspnetcore:2.0
```
Comment (paradise) [inline src/Lykke.Job.BitcoinGold/Dockerfile]: update to 2.1

File: src/Lykke.Job.BitcoinGold/Functions/UpdateObservableOperations.cs [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+                await _log.WriteWarningAsync(nameof(UpdateObservableOperations), nameof(DetectUnconfirmedTransactions),
+                    unconfirmedTransaction.ToJson(), "OperationMeta not found");
+
+                return;
+            }
+
+            var confirmationCount = await _blockChainProvider.GetTxConfirmationCount(unconfirmedTransaction.TxHash);
+
+            var isCompleted = confirmationCount >= _confirmationsSettings.MinConfirmationsToDetectOperation;
+            ;
```
Comment (paradise) [inline src/Lykke.Job.BitcoinGold/Functions/UpdateObservableOperations.cs]: ?

**Review by paradise — APPROVED** [review]

File: src/Lykke.Service.BitcoinGold.API.Services/BlockChainProviders/InsightApi/InsightApiBlockChainProvider.cs [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+
+
+        public async Task<long> GetBalanceSatoshiFromUnspentOutputs(string address, int minConfirmationCount)
+        {
+            var unspentOutputs = (await GetUnspentOutputsResponce(address)).Where(p => p.Confirmation >= minConfirmationCount);
+
+            return unspentOutputs.Sum(p => p.Satoshi);
+        }
+
+        private async Task<IEnumerable<AddressUnspentOutputsResponce>> GetUnspentOutputsResponceBatched(IEnumerable<string> addresses)
```
Comment (PaterSantyago) [inline src/Lykke.Service.BitcoinGold.API.Services/BlockChainProviders/InsightApi/InsightApiBlockChainProvider.cs]: Unused method

File: src/Lykke.Service.BitcoinGold.API.Services/Binder/ServiceModule.cs [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+using Lykke.Service.BitcoinGold.API.Services.Transactions;
+using Lykke.Service.BitcoinGold.API.Services.Wallet;
+using Lykke.SettingsReader;
+using NBitcoin;
+
+namespace Lykke.Service.BitcoinGold.API.Services.Binder
+{
+    public class ServiceModule : Module
+    {
+        private readonly ILog _log;
```
Comment (PaterSantyago) [inline src/Lykke.Service.BitcoinGold.API.Services/Binder/ServiceModule.cs]: Unused field

File: src/Lykke.Service.BitcoinGold.API.Services/LifeiteManagers/ShutdownManager.cs [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+using Lykke.Service.BitcoinGold.API.Core.Services;
+
+namespace Lykke.Service.BitcoinGold.API.Services.LifeiteManagers
+{
+    // NOTE: Sometimes, shutdown process should be expressed explicitly. 
+    // If this is your case, use this class to manage shutdown.
+    // For example, sometimes some state should be saved only after all incoming message processing and 
+    // all periodical handler was stopped, and so on.
+
+    public class ShutdownManager : IShutdownManager
```
Comment (PaterSantyago) [inline src/Lykke.Service.BitcoinGold.API.Services/LifeiteManagers/ShutdownManager.cs]: ShutdownManager does nothing, you can remove it

File: src/Lykke.Service.BitcoinGold.API.Services/LifeiteManagers/StartupManager.cs [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+namespace Lykke.Service.BitcoinGold.API.Services.LifeiteManagers
+{
+    // NOTE: Sometimes, startup process which is expressed explicitly is not just better, 
+    // but the only way. If this is your case, use this class to manage startup.
+    // For example, sometimes some state should be restored before any periodical handler will be started, 
+    // or any incoming message will be processed and so on.
+    // Do not forget to remove As<IStartable>() and AutoActivate() from DI registartions of services, 
+    // which you want to startup explicitly.
+
+    public class StartupManager : IStartupManager
```
Comment (PaterSantyago) [inline src/Lykke.Service.BitcoinGold.API.Services/LifeiteManagers/StartupManager.cs]: StatupManager does nothing, you can remove it

File: src/Lykke.Service.BitcoinGold.API.AzureRepositories/SpentOutputs/SpentOutputRepository.cs [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+        public async Task<IEnumerable<IOutput>> GetSpentOutputs(IEnumerable<IOutput> outputs)
+        {
+            return await _table.GetDataAsync(outputs.Select(o =>
+                new Tuple<string, string>(SpentOutputEntity.GeneratePartitionKey(o.TransactionHash), SpentOutputEntity.GenerateRowKey(o.N))));
+        }
+
+        public async Task RemoveOldOutputs(DateTime bound)
+        {
+            string continuation = null;
+            IEnumerable<SpentOutputEntity> outputs = null;
```
Comment (PaterSantyago) [inline src/Lykke.Service.BitcoinGold.API.AzureRepositories/SpentOutputs/SpentOutputRepository.cs]: Null assignment is not necessary, also, outputs declaration can be moved to the "do" scope

File: src/Lykke.Service.BitcoinGold.API.AzureRepositories/Wallet/ObservableWalletRepositorycs.cs [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+using AzureStorage;
+using Common;
+using Lykke.Service.BitcoinGold.API.Core.Domain.Health.Exceptions;
+using Lykke.Service.BitcoinGold.API.Core.Wallet;
+using Microsoft.WindowsAzure.Storage;
+using Microsoft.WindowsAzure.Storage.Table;
+
+namespace Lykke.Service.BitcoinGold.API.AzureRepositories.Wallet
+{
+    public class ObservableWalletEntity : TableEntity, IObservableWallet
```
Comment (PaterSantyago) [inline src/Lykke.Service.BitcoinGold.API.AzureRepositories/Wallet/ObservableWalletRepositorycs.cs]: Move class to separate file

File: src/Lykke.Service.BitcoinGold.API.AzureRepositories/Wallet/ObservableWalletRepositorycs.cs [hunk 9]
```diff
@@ -0,0 +1,69 @@
...
+                Address = source.Address,
+                PartitionKey = GeneratePartitionKey(source.Address),
+                RowKey = GenerateRowKey(source.Address)
+            };
+        }
+    }
+    public class ObservableWalletRepository : IObservableWalletRepository
+    {
+        private readonly INoSQLTableStorage<ObservableWalletEntity> _storage;
+        private const int EntityExistsHttpStatusCode = 409;
```
Comment (PaterSantyago) [inline src/Lykke.Service.BitcoinGold.API.AzureRepositories/Wallet/ObservableWalletRepositorycs.cs]: Constant is never used

File: src/Lykke.Service.BitcoinGold.API.AzureRepositories/Wallet/ObservableWalletRepositorycs.cs [hunk 10]
```diff
@@ -0,0 +1,69 @@
...
+        private const int EntityNotExistsHttpStatusCode = 404;
```
Comment (PaterSantyago) [inline src/Lykke.Service.BitcoinGold.API.AzureRepositories/Wallet/ObservableWalletRepositorycs.cs]: This one too

File: src/Lykke.Service.BitcoinGold.API/Models/AddressContextContract.cs [hunk 11]
```diff
@@ -0,0 +1,11 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Lykke.Job.BitcoinGold.Models
```
Comment (PaterSantyago) [inline src/Lykke.Service.BitcoinGold.API/Models/AddressContextContract.cs]: Namespace does not correspond to file location

File: src/Lykke.Service.BitcoinGold.API/Modules/BitcoinCashApiModule.cs [hunk 12]
```diff
@@ -0,0 +1,47 @@
...
+using Lykke.Service.BitcoinGold.API.Services.Health;
+using Lykke.Service.BitcoinGold.API.Services.LifeiteManagers;
+using Lykke.SettingsReader;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Lykke.Service.BitcoinGold.API.Modules
+{
+    public class BitcoinGoldApiModule : Module
+    {
+        private readonly IReloadingManager<BitcoinGoldApiSettings> _settings;
```
Comment (PaterSantyago) [inline src/Lykke.Service.BitcoinGold.API/Modules/BitcoinCashApiModule.cs]: Field is never used

File: src/Lykke.Service.BitcoinGold.API.AzureRepositories/Asset/AssetRepository.cs [hunk 13]
```diff
@@ -0,0 +1,42 @@
...
+        };
+
+        public Task<IPaginationResult<IAsset>> GetPaged(int take, string continuation)
+        {
+            return Task.FromResult(PaginationResult<IAsset>.Create(_mockList, null));
+        }
+
+        public Task<IAsset> GetById(string assetId)
+        {
+            return Task.FromResult(_mockList.SingleOrDefault(p=>p.AssetId == assetId));
```
Comment (PaterSantyago) [inline src/Lykke.Service.BitcoinGold.API.AzureRepositories/Asset/AssetRepository.cs:31]: You can use .Signle() here

File: src/Lykke.Service.BitcoinGold.API.AzureRepositories/Asset/AssetRepository.cs [hunk 14]
```diff
@@ -0,0 +1,42 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Lykke.Service.BitcoinGold.API.Core.Asset;
+using Lykke.Service.BitcoinGold.API.Core.Constants;
+using Lykke.Service.BitcoinGold.API.Core.Pagination;
+
+namespace Lykke.Service.BitcoinGold.API.AzureRepositories.Asset
+{
+    public class Asset : IAsset
```
Comment (PaterSantyago) [inline src/Lykke.Service.BitcoinGold.API.AzureRepositories/Asset/AssetRepository.cs]: Move class to separate file

File: src/Lykke.Service.BitcoinGold.API.AzureRepositories/Binder/RepositoryModule.cs [hunk 15]
```diff
@@ -0,0 +1,86 @@
...
+            RegisterBlob(builder);
+        }
+
+        private void RegisterRepo(ContainerBuilder builder)
+        {
+            builder.RegisterInstance(new AssetRepository())
+                .As<IAssetRepository>();
+
+            builder.RegisterInstance(new OperationMetaRepository(
+                AzureTableStorage<OperationMetaEntity>.Create(_settings.Nested(p => p.Db.DataConnString),
```
Comment (PaterSantyago) [inline src/Lykke.Service.BitcoinGold.API.AzureRepositories/Binder/RepositoryModule.cs:43]: It's better to add static Create method to the repository and incapsulate table name and other repo-creation logic inside repo

File: src/Lykke.Service.BitcoinGold.API.AzureRepositories/Operations/OperationEventRepository.cs [hunk 16]
```diff
@@ -0,0 +1,64 @@
+﻿using System;
+using System.Threading.Tasks;
+using AzureStorage;
+using Lykke.Service.BitcoinGold.API.Core.Operation;
+using Microsoft.WindowsAzure.Storage.Table;
+
+namespace Lykke.Service.BitcoinGold.API.AzureRepositories.Operations
+{
+    public class OperationEventTableEntity : TableEntity, IOperationEvent
```
Comment (PaterSantyago) [inline src/Lykke.Service.BitcoinGold.API.AzureRepositories/Operations/OperationEventRepository.cs]: Move to separate file

File: src/Lykke.Service.BitcoinGold.API.AzureRepositories/Operations/OperationEventRepository.cs [hunk 17]
```diff
@@ -0,0 +1,64 @@
...
+    public class OperationEventRepository : IOperationEventRepository
+    {
+        private readonly INoSQLTableStorage<OperationEventTableEntity> _storage;
+
+        public OperationEventRepository(INoSQLTableStorage<OperationEventTableEntity> storage)
+        {
+            _storage = storage;
+        }
+
+        public async Task InsertIfNotExist(IOperationEvent operationEvent)
```
Comment (PaterSantyago) [inline src/Lykke.Service.BitcoinGold.API.AzureRepositories/Operations/OperationEventRepository.cs:17]: InsertIfNotExistAsync

File: src/Lykke.Service.BitcoinGold.API.AzureRepositories/Operations/OperationEventRepository.cs [hunk 18]
```diff
@@ -0,0 +1,64 @@
...
+        {
+            await _storage.CreateIfNotExistsAsync(OperationEventTableEntity.Create(operationEvent));
+        }
+
+        public async Task<bool> Exist(Guid operationId, OperationEventType type)
```
Comment (PaterSantyago) [inline src/Lykke.Service.BitcoinGold.API.AzureRepositories/Operations/OperationEventRepository.cs:22]: ExistAsync

File: src/Lykke.Service.BitcoinGold.API.AzureRepositories/Operations/OperationMetaRepository.cs [hunk 19]
```diff
@@ -0,0 +1,84 @@
+﻿using System;
+using System.Threading.Tasks;
+using AzureStorage;
+using Common;
+using Lykke.Service.BitcoinGold.API.Core.Operation;
+using Microsoft.WindowsAzure.Storage.Table;
+
+namespace Lykke.Service.BitcoinGold.API.AzureRepositories.Operations
+{
+    public class OperationMetaEntity : TableEntity, IOperationMeta
```
Comment (PaterSantyago) [inline src/Lykke.Service.BitcoinGold.API.AzureRepositories/Operations/OperationMetaRepository.cs]: Move to separate file

File: src/Lykke.Service.BitcoinGold.API.AzureRepositories/Operations/OperationMetaRepository.cs [hunk 20]
```diff
@@ -0,0 +1,84 @@
...
+            }
+
+            public static OperationMetaEntity Create(IOperationMeta source)
+            {
+                return Map(GeneratePartitionKey(source.OperationId), GenerateRowKey(source.OperationId), source);
+            }
+        }
+    }
+
+    public class OperationMetaRepository:IOperationMetaRepository
```
Comment (PaterSantyago) [inline src/Lykke.Service.BitcoinGold.API.AzureRepositories/Operations/OperationMetaRepository.cs]: Add Async suffixes

File: src/Lykke.Service.BitcoinGold.API.AzureRepositories/SpentOutputs/SpentOutputRepository.cs [hunk 21]
```diff
@@ -0,0 +1,76 @@
...
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using AzureStorage;
+using Lykke.Service.BitcoinGold.API.Core.TransactionOutputs;
+using Microsoft.WindowsAzure.Storage.Table;
+
+namespace Lykke.Service.BitcoinGold.API.AzureRepositories.SpentOutputs
+{
+    public class SpentOutputEntity : TableEntity, IOutput
```
Comment (PaterSantyago) [inline src/Lykke.Service.BitcoinGold.API.AzureRepositories/SpentOutputs/SpentOutputRepository.cs]: Move to separate file

File: src/Lykke.Service.BitcoinGold.API.AzureRepositories/SpentOutputs/SpentOutputRepository.cs [hunk 22]
```diff
@@ -0,0 +1,76 @@
...
+
+        public static string GeneratePartitionKey(string transactionHash)
+        {
+            return transactionHash;
+        }
+    }
+
+
+
+    public class SpentOutputRepository : ISpentOutputRepository
```
Comment (PaterSantyago) [inline src/Lykke.Service.BitcoinGold.API.AzureRepositories/SpentOutputs/SpentOutputRepository.cs]: Add Async suffixes

File: src/Lykke.Service.BitcoinGold.API.AzureRepositories/Transactions/ObservableOperationRepository.cs [hunk 23]
```diff
@@ -0,0 +1,95 @@
+﻿using System;
+using System.Threading.Tasks;
+using AzureStorage;
+using Common;
+using Lykke.Service.BitcoinGold.API.Core.ObservableOperation;
+using Microsoft.WindowsAzure.Storage.Table;
+
+namespace Lykke.Service.BitcoinGold.API.AzureRepositories.Transactions
+{
+    public class ObservableOperationEntity : TableEntity, IObservableOperation
```
Comment (PaterSantyago) [inline src/Lykke.Service.BitcoinGold.API.AzureRepositories/Transactions/ObservableOperationRepository.cs]: Move to separate file

File: src/Lykke.Service.BitcoinGold.API.AzureRepositories/Transactions/ObservableOperationRepository.cs [hunk 24]
```diff
@@ -0,0 +1,95 @@
...
+            }
+
+            public static ObservableOperationEntity Create(IObservableOperation source)
+            {
+                return Map(GeneratePartitionKey(source.OperationId), GenerateRowKey(source.OperationId), source);
+            }
+        }
+    }
+
+    public class ObservableOperationRepository : IObservableOperationRepository
```
Comment (PaterSantyago) [inline src/Lykke.Service.BitcoinGold.API.AzureRepositories/Transactions/ObservableOperationRepository.cs]: Add async suffixes

File: src/Lykke.Service.BitcoinGold.API.AzureRepositories/Transactions/TransactionBlobStorage.cs [hunk 25]
```diff
@@ -0,0 +1,41 @@
+﻿using System;
+using System.Text;
+using System.Threading.Tasks;
+using AzureStorage;
+using Lykke.Service.BitcoinGold.API.Core.Transactions;
+
+namespace Lykke.Service.BitcoinGold.API.AzureRepositories.Transactions
+{
+    public class TransactionBlobStorage : ITransactionBlobStorage
```
Comment (PaterSantyago) [inline src/Lykke.Service.BitcoinGold.API.AzureRepositories/Transactions/TransactionBlobStorage.cs:9]: Add Async suffixes

File: src/Lykke.Service.BitcoinGold.API.AzureRepositories/Transactions/UnconfirmedTransactionRepository.cs [hunk 26]
```diff
@@ -0,0 +1,66 @@
...
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using AzureStorage;
+using Common;
+using Lykke.Service.BitcoinGold.API.Core.Transactions;
+using Microsoft.WindowsAzure.Storage.Table;
+
+namespace Lykke.Service.BitcoinGold.API.AzureRepositories.Transactions
+{
+    public class UnconfirmedTransactionEntity : TableEntity, IUnconfirmedTransaction
```
Comment (PaterSantyago) [inline src/Lykke.Service.BitcoinGold.API.AzureRepositories/Transactions/UnconfirmedTransactionRepository.cs]: Move to separate file

File: src/Lykke.Service.BitcoinGold.API.AzureRepositories/Transactions/UnconfirmedTransactionRepository.cs [hunk 27]
```diff
@@ -0,0 +1,66 @@
...
+            {
+                PartitionKey = GeneratePartitionKey(source.OperationId),
+                RowKey = GenerateRowKey(source.OperationId),
+                OperationId = source.OperationId,
+                TxHash = source.TxHash
+            };
+        }
+    }
+
+    public class UnconfirmedTransactionRepository: IUnconfirmedTransactionRepository
```
Comment (PaterSantyago) [inline src/Lykke.Service.BitcoinGold.API.AzureRepositories/Transactions/UnconfirmedTransactionRepository.cs]: Add Async suffixes

File: src/Lykke.Service.BitcoinGold.API.AzureRepositories/Wallet/ObservableWalletRepositorycs.cs [hunk 28]
```diff
@@ -0,0 +1,69 @@
...
+        {
+            return new ObservableWalletEntity
+            {
+                Address = source.Address,
+                PartitionKey = GeneratePartitionKey(source.Address),
+                RowKey = GenerateRowKey(source.Address)
+            };
+        }
+    }
+    public class ObservableWalletRepository : IObservableWalletRepository
```
Comment (PaterSantyago) [inline src/Lykke.Service.BitcoinGold.API.AzureRepositories/Wallet/ObservableWalletRepositorycs.cs]: Add async suffixes

File: src/Lykke.Service.BitcoinGold.API.AzureRepositories/Wallet/WalletBalanceRepository.cs [hunk 29]
```diff
@@ -0,0 +1,69 @@
...
+using System.Threading.Tasks;
+using AzureStorage;
+using Common;
+using Lykke.Service.BitcoinGold.API.Core.Pagination;
+using Lykke.Service.BitcoinGold.API.Core.Wallet;
+using Microsoft.WindowsAzure.Storage.Table;
+
+namespace Lykke.Service.BitcoinGold.API.AzureRepositories.Wallet
+{
+    public class WalletBalanceEntity : TableEntity, IWalletBalance
```
Comment (PaterSantyago) [inline src/Lykke.Service.BitcoinGold.API.AzureRepositories/Wallet/WalletBalanceRepository.cs]: Move to separate file

File: src/Lykke.Service.BitcoinGold.API.AzureRepositories/Wallet/WalletBalanceRepository.cs [hunk 30]
```diff
@@ -0,0 +1,69 @@
...
+                BalanceSatoshi = source.BalanceSatoshi,
+                RowKey = GenerateRowKey(source.Address),
+                PartitionKey = GeneratePartitionKey(source.Address),
+                Updated = source.Updated,
+                UpdatedAtBlockHeight = source.UpdatedAtBlockHeight
+            };
+        }
+    }
+
+    public class WalletBalanceRepository : IWalletBalanceRepository
```
Comment (PaterSantyago) [inline src/Lykke.Service.BitcoinGold.API.AzureRepositories/Wallet/WalletBalanceRepository.cs]: Add Async suffixes

File: src/Lykke.Service.BitcoinGold.API.Core/Domain/Health/HealthIssue.cs [hunk 31]
```diff
@@ -0,0 +1,17 @@
+﻿namespace Lykke.Service.BitcoinGold.API.Core.Domain.Health
+{
+    public class HealthIssue
```
Comment (PaterSantyago) [inline src/Lykke.Service.BitcoinGold.API.Core/Domain/Health/HealthIssue.cs]: Heart issues has been moved to Common library

File: src/Lykke.Service.BitcoinGold.API.Core/Domain/Health/HealthIssuesCollection.cs [hunk 32]
```diff
@@ -0,0 +1,32 @@
+using System.Collections;
+using System.Collections.Generic;
+
+namespace Lykke.Service.BitcoinGold.API.Core.Domain.Health
+{
+    public class HealthIssuesCollection : IReadOnlyCollection<HealthIssue>
```
Comment (PaterSantyago) [inline src/Lykke.Service.BitcoinGold.API.Core/Domain/Health/HealthIssuesCollection.cs]: HealthIssuesCollection has been moved to common library

File: src/Lykke.Service.BitcoinGold.API.Services/Broadcast/BroadcastService.cs [hunk 33]
```diff
@@ -0,0 +1,92 @@
...
+
+            var lastBlockHeight = await _blockChainProvider.GetLastBlockHeight();
+        
+            await _blockChainProvider.BroadCastTransaction(tx);
+
+            await _observableOperationRepository.InsertOrReplace(ObervableOperation.Create(operation, BroadcastStatus.InProgress, hash, lastBlockHeight));
+
+            await _unconfirmedTransactionRepository.InsertOrReplace(UnconfirmedTransaction.Create(operationId, hash));
+
+            await _operationEventRepository.InsertIfNotExist(OperationEvent.Create(operationId, OperationEventType.Broadcasted));
```
Comment (PaterSantyago) [inline src/Lykke.Service.BitcoinGold.API.Services/Broadcast/BroadcastService.cs]: I think, I'll be better, to perform this operation last.

File: src/Lykke.Service.BitcoinGold.API.Services/Operations/OperationService.cs [hunk 34]
```diff
@@ -0,0 +1,70 @@
...
+            BitcoinAddress toAddress,
+            string assetId,
+            Money amountToSend,
+            bool includeFee)
+        {
+            var existingOperation = await _operationMetaRepository.Get(operationId);
+            if (existingOperation != null)
+                return await GetExistingTransaction(existingOperation.OperationId, existingOperation.Hash);
+
+            var buildedTransaction = await _transactionBuilder.GetTransferTransaction(fromAddress, fromAddressPubkey, toAddress, amountToSend, includeFee);
```
Comment (PaterSantyago) [inline src/Lykke.Service.BitcoinGold.API.Services/Operations/OperationService.cs]: Not builded but built

File: src/Lykke.Service.BitcoinGold.API/Startup.cs [hunk 35]
```diff
@@ -0,0 +1,215 @@
...
+            var dbLogConnectionString = dbLogConnectionStringManager.CurrentValue;
+
+            // Creating azure storage logger, which logs own messages to concole log
+            if (!string.IsNullOrEmpty(dbLogConnectionString) && !(dbLogConnectionString.StartsWith("${") && dbLogConnectionString.EndsWith("}")))
+            {
+                var persistenceManager = new LykkeLogToAzureStoragePersistenceManager(
+                    AzureTableStorage<LogEntity>.Create(dbLogConnectionStringManager, "BitcoinGoldApiLog", consoleLogger),
+                    consoleLogger);
+
+                var slackNotificationsManager = new LykkeLogToAzureSlackNotificationsManager(slackService, consoleLogger);
```
Comment (PaterSantyago) [inline src/Lykke.Service.BitcoinGold.API/Startup.cs]: Add logging of errors to special slack channel, like it done here https://github.com/LykkeCity/Lykke.Job.BlockchainCashoutProcessor/blob/f20fa806a96ceb19a08ddf3514d54c1bf831db3b/src/Lykke.Job.BlockchainCashoutProcessor/Startup.cs#L227-L244

**Review by PaterSantyago — CHANGES_REQUESTED** [review]
Added some code-style comments:
1) Async suffixed of async methods lost
2) It's better to place classes in separate files
3) Build it irregular verb, so, not builded, but built
4) There some unnecessary 'using'`s all over the code

Also added some non code-style comments

File: src/Lykke.Service.BitcoinGold.API.AzureRepositories/Asset/AssetRepository.cs (hunk 13 above)
Comment (troshkin) [reply src/Lykke.Service.BitcoinGold.API.AzureRepositories/Asset/AssetRepository.cs:31]: Single() will throw error code 500 in auto tests

**Pull request merged**